IdentityCore/src/broker_operation/request/token_request/MSIDBrokerOperationBrowserTokenRequest.m

```diff
@@ -54,6 +57,18 @@ - (instancetype)initWithRequest:(NSURL *)requestURL
         }
         
         _requestURL = requestURL;
+
+        if (!isInterativeRequest && [requestValidator isMSALRequest:headers[PLATFORM_KEY]])
+        {
+            if (error)
+            {
+                NSString *errorMessage = [NSString stringWithFormat:@"Failed to create operation request, application with SKU='%@' is not allowed to make an SSO silent browser request", headers[PLATFORM_KEY]];
+                *error = MSIDCreateError(MSIDErrorDomain,MSIDErrorInvalidInternalParameter,errorMessage,nil, nil, nil, nil, nil, YES);
```

Comment:
Looking at the implementation in our test extension, when we return error when creating SSO request, scenario will just hang:
> guard let ssoRequest = try? ADBrokerSSOExtensionRequest(ssoExtensionRequest: request, ssoExtensionMode: ssoExtensionMode, ssoModeForBrowser: ssoBrowserMode) else
        {
            return
        }

Also, do we need this to be in the common core layer at all? This seems like something that belongs to the broker layer so maybe it can handled entirely inside ADBrokerSSOExtensionRequest?

Reply:
good point looking into this